### PR TITLE
*: remove raft module

### PR DESCRIPTION
**Pull request by BusyJay**
Now that raft becomes a standalone crate, raft module is not maintained anymore, let's remove it.

### CODE REVIEW

**Comment by siddontang** [comment]
is it compatible with other TiKV?

**Comment by BusyJay** [comment]
Yes.

**Comment by siddontang** [comment]
do we need to unpublic the raft fields and use public API instead here?

**Comment by BusyJay** [comment]
Yes. It will be done before 1.0 is released.

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -1391,7 +1392,7 @@ impl Peer {
         req: RaftCmdRequest,
         metrics: &mut RaftProposeMetrics,
     ) -> Result<u64> {
-        if self.raft_group.raft.pending_conf {
+        if self.raft_group.raft.pending_conf_index > self.get_store().applied_index() {
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/peer.rs]: The raft repo have changed?

File: Cargo.lock [hunk 2]
```diff
@@ -1217,11 +1256,30 @@ name = "winapi"
 version = "0.2.8"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 
+[[package]]
+name = "winapi"
```
Comment (zhangjinpeng1987) [inline Cargo.lock]: Where `winapi` will be used?

File: src/raftstore/store/peer.rs (hunk 1 above)
Comment (BusyJay) [reply src/raftstore/store/peer.rs]: Yes. Several fixes are merged.

File: Cargo.lock (hunk 2 above)
Comment (BusyJay) [reply Cargo.lock]: It's used for the `rand` crate.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
Please also approve pingcap/kvproto#231.

**Comment by BusyJay** [comment]
PTAL, /run-integration-tests

**Comment by Hoverbear** [comment]
🎉

**Pull request merged**